### DISCUSSION TITLE
RTSP camera auto start not working after server restart

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#5011] [FAQs] [answered]
Multiple RTSP cameras are broadcasting on my server, but when I restart the server, the streams do not automatically restart. I had to manually start the streams. Is there a way to fix this issue?

**Comment by asharamseervi** [comment, marked as answer]
In AMS v2.5.3 and later, the stream auto-fetcher is disabled by default so streams do not start themselves after a server restart. Change the following flag to true in the Advance application settings so that streams start fetching automatically.

`"startStreamFetcherAutomatically": true,`

The streams should now begin to function automatically.

**Reply by klasadoroty** [reply]
Hi. I'm not proficient in this... but a question like this: does it matter which line we put the entry in?
Where exactly to put it?
Thank you in advance.

**Reply by yashtandon113** [reply]
@klasadoroty 
You can put the property/settings in any line under /usr/local/antmedia/webapps/app-name/WEB-INF/red5-web.properties.

**Comment by klasadoroty** [comment]
Thank you for your reply. It does work.

But, how to secure the publisher?
Publisher IP Filter - this does not seem to work.
How to do it properly?
It should work with Community Edition? 
Can you give in a practical, simple way: how to secure the publisher?
Currently, anyone will connect to AMS, without problems.

**Reply by yashtandon113** [reply]
@klasadoroty
Check out https://antmedia.io/docs/guides/developer-sdk-and-api/rest-api-guide/stream-security/ for more information about Stream Security.

The Stream Security feature is only available in AMS Enterprise Edition. Ant Media Server is available for a five-day free trial on the AWS marketplace. Check out https://www.youtube.com/watch?v=EH6v-yUyzjU for more information.

You can also get a 14-day self hosted trial license from https://antmedia.io/free-trial/.

**Comment by klasadoroty** [comment]
Thanks.
Hash-Based Token - is only available in AMS Enterprise Edition?
Publisher IP Filter - is only available in AMS Enterprise Edition?
... no security in the non-commercial version?

**Reply by yashtandon113** [reply]
@klasadoroty 
Yes, there is no stream security mechanism in the Community Edition.

You can disable the **Accept Undefined Streams** option so that not every stream is published. Only pre-configured streams in the application can be published.

**Reply by klasadoroty** [reply]
Thanks.
I understand.
So AMS in such a version is ... just a toy. It is not suitable for any application.
You can only see that it works.
It makes absolutely no sense.
What is it supposed to be used for?
If I want to test it, I will take 14 days to try it.
Example: at our church we want to launch a transmission. Once a week. If the transmission will not be active then ... some idiot will launch his transmission , for example, with offensive content.
In my opinion, security for broadcasting should be.
I understand, the other security only for a fee.

**Reply by yashtandon113** [reply]
@klasadoroty 

While the Community version of AMS does not provide all the features available in the Enterprise editions, it is still a fully functional video solution that allows users to publish, play, and record streams. That being said, the Enterprise editions offer additional benefits such as WebRTC support with 0.5 seconds of latency, improved stream security features and many more. The choice of which version to use would depend on the specific requirements of your application, and we recommend evaluating the different editions to determine which one best fits your needs.

You can also check feature comparison between Community Edition and Enterprise edition on below link:
https://antmedia.io/docs/#community--enterprise-edition-comparison

I hope this helps!

**Reply by klasadoroty** [reply]
Well... it didn't help.
I understand what you wrote.
Nonetheless, I believe that simple security to the access point MUST BE.
Without it, the software is only good for playing with.
And I think that's what you mean.
But at this point the cost is comparable to others. 
Only those others offer in paid versions much more.
Overall, AMS is interesting. Probably in the Enterprice version.

**Reply by klasadoroty** [reply]
What you offer in this version is not logical.
Example: you give the possibility of security for iframes.
What for? In the Community version ... meaningless.
And no security for streaming.

**Reply by yashtandon113** [reply]
@klasadoroty 
Yes, it is possible that the Community Edition will not meet your needs, in which case you might consider the Enterprise Edition.

There are further methods for restricting traffic on your server firewall. For example, if you're intending to use RTMP to publish streams, only enable TCP Port 1935 for a certain publisher's public IP address and block it for everyone else.

**Reply by klasadoroty** [reply]
I did. It's Ok.
But I still find an approach like yours not very logical.
Thank You.
Best Regards

**Reply by mekya** [reply]
Hi @klasadoroty,
I appreciate your involvement into this conversation. 

Just let me give some comments and try to make you happy. 

Our idea is that if anyone want to have secure streaming then it's something premium and it requires enterprise edition. 
For community edition, as you mention, yes it's like a toy for testing and doing some hobby projects. 

Lastly, if you need enterprise edition and you're from a church/education institute that you don't have budget for this, let us know. 
We can help you ;)

This is related to one of the item in the manifesto
> Always seek win-win relationships. If a user can’t afford a license, let’s find alternative win-win solutions.

 https://antmedia.io/about-us/

Cheers

**Reply by klasadoroty** [reply]
Hi, Thank you for the information.
Community edition is a fully working server, that's my understanding.
There is only no premium feature. 
In my opinion it works very well.
We just don't know how it will perform with a larger application - 1000 - 4000 recipients.
But we will test that too.
At the moment we are very satisfied.
..And we have a comparison with others.... e.g. wowza.
Best Regards

**Reply by mekya** [reply]
Thank you for your response @klasadoroty 

It's our pleasure to help you. 

Why don't we have some coffee together as mentioned in the about us? 
https://antmedia.io/about-us/

Cheers
